core/backend/src/IModelTransformer.ts

```diff
@@ -746,46 +749,75 @@ export class IModelTransformer extends IModelExportHandler {
     return targetElementAspectProps;
   }
 
+  /** The directory where schemas will be exported, a random temporary directory */
+  protected _schemaExportDir: string = path.join(KnownLocations.tmpdir, Guid.createValue());
+
+  /** Export a schema that was deserialized to a given path */
+  private shouldExportDeserializedSchema(schemaPath: string): boolean {
+    let schemaSrc: string;
+    try {
+      schemaSrc = IModelJsFs.readFileSync(schemaPath).toString("utf8");
+    } catch(err) {
+      Logger.logError(loggerCategory, `error reading xml schema file at '${schemaPath}`);
+      return true;
+    }
+    // ad-hoc fast xml parsing
+    const schemaVersionMatch = /<ECSchema .*?version="([0-9.]+)"/.exec(schemaSrc);
+    const schemaNameMatch = /<ECSchema .*?schemaName="([^"]+)"/.exec(schemaSrc);
+    if (schemaVersionMatch == null || schemaNameMatch == null) {
+      Logger.logError(loggerCategory, `failed to parse schema name or version, first 200 chars: '${schemaSrc.slice(0, 200)}'`);
+      return true;
+    }
+    const [_fullVersionMatch, versionString] = schemaVersionMatch;
+    const [_fullNameMatch, schemaName] = schemaNameMatch;
+    return this.shouldExportSchemaCheck(schemaName, Schema.toSemverString(versionString));
+  }
+
+  /** given a schema identified just by name and the version string, check if it should be imported into the target */
+  private shouldExportSchemaCheck(schemaName: string, versionInSource: string): boolean {
+    const versionInTarget = this.targetDb.querySchemaVersion(schemaName);
+    if (versionInTarget === undefined)
+      return true;
+    return Semver.gt(versionInSource, versionInTarget);
+  }
+
+  /** Override of [IModelExportHandler.shouldExportSchema]($backend) that is called to determine if a schema should be exported
+   * @note the default behavior doesn't import schemas older than those already in the target
+   */
+  protected shouldExportSchema(schema: ECSchemaMetaData.Schema): boolean {
+    const versionInSource = `${schema.readVersion}.${schema.writeVersion}.${schema.minorVersion}`;
+    return this.shouldExportSchemaCheck(schema.name, versionInSource);
+  }
+
+  /** Override of [IModelExportHandler.onExportSchema]($backend) that imports a schema into the target iModel when it is exported from the source iModel. */
+  protected async onExportSchema(schema: ECSchemaMetaData.Schema): Promise<void> {
+    const schemaPath = path.join(this._schemaExportDir, `${schema.fullName}.ecschema.xml`);
+    const xmlDoc = new DOMParser().parseFromString(`<?xml version="1.0" encoding="UTF-8"?>`, "application/xml");

```

Comment:
Is there a way to just write the schema XML file out without parsing it? Core code produced it so it should be valid plus the XML will be validated in native code before importing.

Reply:
that isn't actually parsing the schema. It's parsing just the xml header string: `<?xml version="1.0" encoding="UTF-8"?>`
the `schema.toXml` then mutates that document object, without any parsing. Maybe I don't even need to parse the xml-header but it seemed the best way to make sure the document contains the xml header after being filled.

Reply:
going to replace with an added alpha API method `ECSchemaMetaData.Schema.toXmlString`